app/web_framework/cluster_master_application.py

```diff
@@ -30,38 +30,71 @@ def __init__(self, cluster_master):
+                                    ])
+                                ]),
+                                RouteNode(r'result', _SubjobResultHandler)
                             ])
                         ])
+                    ])
+                ]),
+                RouteNode(r'queue', _QueueHandler),
+                RouteNode(r'slave', _SlavesHandler, 'slaves').add_children([
+                    RouteNode(r'(\d+)', _SlaveHandler, 'slave').add_children([
+                        RouteNode(r'shutdown', _SlaveShutdownHandler, 'shutdown')
                     ]),
-                    RouteNode(r'queue', _QueueHandler),
-                    RouteNode(r'slave', _SlavesHandler, 'slaves').add_children([
-                        RouteNode(r'(\d+)', _SlaveHandler, 'slave').add_children([
-                            RouteNode(r'shutdown', _SlaveShutdownHandler, 'shutdown')
-                        ]),
-                        RouteNode(r'shutdown', _SlavesShutdownHandler, 'shutdown')
-                    ]),
-                    RouteNode(r'eventlog', _EventlogHandler)
+                    RouteNode(r'shutdown', _SlavesShutdownHandler, 'shutdown')
+                ]),
+                RouteNode(r'eventlog', _EventlogHandler)
+            ])]
+
+        api_v2 = [
+            RouteNode(r'metrics', _MetricsHandler),
+            RouteNode(r'version', _VersionHandler),
+            RouteNode(r'builds', _BuildsHandler, 'builds').add_children([
+                RouteNode(r'(\d+)', _BuildHandler, 'build').add_children([
+                    RouteNode(r'result', _BuildResultRedirectHandler),
+                    RouteNode(r'artifacts.tar.gz', _BuildTarResultHandler),
+                    RouteNode(r'artifacts.zip', _BuildZipResultHandler),
+                    RouteNode(r'subjob', _SubjobsHandler, 'subjobs').add_children([
+                        RouteNode(r'(\d+)', _SubjobHandler, 'subjob').add_children([
+                            RouteNode(r'atom', _AtomsHandler, 'atoms').add_children([
+                                RouteNode(r'(\d+)', _AtomHandler, 'atom').add_children([
+                                    RouteNode(r'console', _AtomConsoleHandler)
+                                ])
+                            ]),
+                            RouteNode(r'result', _SubjobResultHandler)
+                        ])
+                    ])
                 ])
-            ])
+            ]),
+            RouteNode(r'queue', _QueueHandler),
+            RouteNode(r'slave', _SlavesHandler, 'slaves').add_children([
+                RouteNode(r'(\d+)', _SlaveHandler, 'slave').add_children([
+                    RouteNode(r'shutdown', _SlaveShutdownHandler, 'shutdown')
+                ]),
+                RouteNode(r'shutdown', _SlavesShutdownHandler, 'shutdown')
+            ]),
+            RouteNode(r'eventlog', _EventlogHandler)]
+
+        root_route = RouteNode(r'/', _RootHandler)
+        root_route.add_children(api_v1, version=1)
```

Comment:
Do we want to add v1 routes under `/`? And yeah as of now there are no collisions but there would be in the future (or if we want to add v1 routes under `/` as well)